libraries/entities/src/ShapeEntityItem.cpp

```diff
@@ -51,6 +51,14 @@ namespace entity {
     }
 }
 
+// shapeCalculator is a hook for external code that knows how to configure a ShapeInfo
+// for given entity::Shape and dimensions
+ShapeEntityItem::ShapeInfoCalculator shapeCalculator = nullptr;
+
+void ShapeEntityItem::setShapeInfoCalulator(ShapeEntityItem::ShapeInfoCalculator callback) {
+    shapeCalculator = callback;
+}
```

Comment:
This is the trick: we provide a callback that expects to be hooked up with a function that knows how to do the work we need.  The implementation of that function must live in a context that knows about:  `ShapeInfo`, `entity::Shape`, and the `GeometryCache` API.  Which means... it probably needs to be implemented in the interface code itself where all of this comes together.  The interface's `Application` class could register the callback in its initialization.

Reply:
The "work we need" is to identify the correct set of raw local points in `GeometryCache` that correspond to the `entity::Shape` and to copy those (with proper scaling as per the dimensions) into the `ShapeInfo`.

Finally we might need to update `ShapeInfo` to properly compute its hash when the `ShapeManager` uses it.  Probably not but we should audit that code just to make sure it looks like it is doing the right thing.  The idea here is that once the `ShapeInfo` has been populated with everthing it needs it can compute a unique "key" that is used to store the actual `btCollisionShape` in the `ShapeManager`.  Should a second instance of the shape be requested from the `ShapeManager` it would find it in its cache and supply a pointer to the first.

Reply:
Ok, thanks! I believe that I've been in that code before for a different matter.  I'll let you know if I have any questions or run into any issues.

Reply:
Just saw your follow up comment.  I don't know if you had the chance to review the changes from the https://github.com/1P-Cusack/hifi/pull/1; however, I think that the changes there align with what you're describing.  I'll have to refresh my memory regarding the hash generation; however, I may have added a todo/note there.